From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 30 Nov 2022 16:54:28 +0100
Subject: [PATCH] Skip cloning advancement criteria

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"Skip cloning loot parameters" [sic]
By: etil2jz <81570777+etil2jz@users.noreply.github.com>
As part of: Mirai (https://github.com/etil2jz/Mirai)
Licensed under: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/net/minecraft/advancements/Advancement.java b/net/minecraft/advancements/Advancement.java
index b8afa8f7e3a51c92deb4e2d105031ee9eb39e663..7729201c03be1063f7026b106911897e5084147b 100644
--- a/net/minecraft/advancements/Advancement.java
+++ b/net/minecraft/advancements/Advancement.java
@@ -72,7 +72,7 @@ public record Advancement(
         final AdvancementRequirements requirements,
         final boolean sendsTelemetryEvent
     ) {
-        this(parent, display, rewards, Map.copyOf(criteria), requirements, sendsTelemetryEvent, display.map(Advancement::decorateName));
+        this(parent, display, rewards, criteria, requirements, sendsTelemetryEvent, display.map(Advancement::decorateName)); // Gale - Mirai - skip cloning advancement criteria
     }
 
     private static DataResult<Advancement> validate(final Advancement advancement) {
